client/signup/steps/domains/index.jsx

```diff
@@ -577,6 +583,24 @@ export class RenderDomainsStep extends Component {
 		registration.item_subtotal_integer = ( suggestion.sale_cost ?? suggestion.raw_price ) * 100;
 
 		if ( shouldUseMultipleDomainsInCart( this.props.flowName ) ) {
+			if (
+				! this.state.temporaryCart ||
+				! this.state.temporaryCart.some(
+					( domainInCart ) => domainInCart.meta === suggestion.domain_name
+				)
+			) {
+				this.setState( {
+					temporaryCart: [
+						...( this.state.temporaryCart || [] ),
+						{
+							meta: suggestion.domain_name,
+							item_subtotal_display: '...',
+							temporary: true,
+						},
+					],
+				} );
```

Comment:
Can you do setState(prevstate => newstate) here or does this.setState not support that? Wondering if that's part of the issue with the click add on multiple domains fast thing.

Reply:
Til about it @lsl. [Here's more details](https://legacy.reactjs.org/docs/state-and-lifecycle.html) for future reference.

[Added that change here](https://github.com/Automattic/wp-calypso/pull/84171/commits/fbb12747af49b119585e9ca33a4534b23ffabeb7), but unfortunately it didn't change the outcome since the `this.propos.cart.products` doesn't get updated that fast still. Here's more context of testing it: p1700055110866429/1699995887.885749-slack-CKZHG0QCR

Does it make sense to update `this.propos.cart.products` before the POST `/no-cart` call?

Reply:
> Does it make sense to update this.propos.cart.products before the POST /no-cart call?

I think [I found a solution for it](https://github.com/Automattic/wp-calypso/pull/84269).